connor/database/db.go

```diff
@@ -0,0 +1,369 @@
+	_, err := d.connect.Exec(updateDeployStatusDeal, deployStatus, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func (d *Database) UpdateBadGayStatusInPoolDB(id int64, badGuy int64, timeUpdate time.Time) error {
+	_, err := d.connect.Exec(updateStatusPoolDB, badGuy, timeUpdate, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Database) UpdateChangeRequestStatusDealDB(id int64, status sonm.ChangeRequestStatus, timeUpdate time.Time) error {
+	_, err := d.connect.Exec(updateCRStatusDeal, status, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Database) UpdateReportedHashratePoolDB(id string, reportedHashrate float64, timeUpdate time.Time) error {
+	_, err := d.connect.Exec(updateReportedHashrate, reportedHashrate, timeUpdate, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func (d *Database) UpdateAvgPoolDB(id string, avgHashrate float64, timeUpdate time.Time) error {
+	_, err := d.connect.Exec(updateAvgPool, avgHashrate, timeUpdate, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Database) UpdateBanStatusBlackListDB(masterID string, banStatus int64) error {
+	_, err := d.connect.Exec(updateBlackList, banStatus, masterID)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Database) UpdateIterationPoolDB(iteration int64, id int64) error {
+	_, err := d.connect.Exec(updateIterationPool, iteration, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func (d *Database) SetDestroyDealPoolDB(status int64, id int64) error {
+	_, err := d.connect.Exec(setDestroyStatusDeal, status, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Database) GetCountFromDB() (counts int64, err error) {
+	rows, err := d.connect.Query(getCountFromDb)
+	defer rows.Close()
+	if err != nil {
+		return 0, err
+	}
+	for rows.Next() {
+		var count int64
+		err = rows.Scan(&count)
+		if err != nil {
+			return 0, err
+		}
+		return count, nil
+	}
+	return 0, fmt.Errorf("")
+}
+func (d *Database) GetLastActualStepFromDb() (float64, error) {
+	rows, err := d.connect.Query(getLastActualStep)
+	defer rows.Close()
+	if err != nil {
+		return 0, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var max float64
+		err = rows.Scan(&max)
+		if err != nil {
+			return 0, err
+		}
+		return max, nil
+	}
+	return 0, nil
+}
+func (d *Database) GetOrdersFromDB() ([]*OrderDb, error) {
+	rows, err := d.connect.Query(getOrders)
+	defer rows.Close()
+	if err != nil {
+		return nil, err
+	}
+	orders := make([]*OrderDb, 0)
+	for rows.Next() {
+		order := new(OrderDb)
+		err := rows.Scan(&order.OrderID, &order.Price, &order.Hashrate, &order.StartTime, &order.ButterflyEffect, &order.ActualStep)
+		if err != nil {
+			return nil, err
+		}
+		orders = append(orders, order)
+	}
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+	return orders, err
+}
+func (d *Database) GetBlacklistFromDb(failSupplierID string) (string, error) {
```

Comment:
Not really obvious what is happening here, so here are some questions:
1. Could this method return a (bool, error)?
2. What's the convention for the returned string?
3. What is the loop inside the body of the function doing? It looks like only the first iteration of it will work